### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by shaorray**
Three functions often use in gene count table normalization, quantile normalization is very common in ChIP sample, size factor and median ratio are for RNA expression.

### CODE REVIEW

**Comment by cnluzon** [comment]
Hi @shaorray ! Thanks for contributing to the repo! :) I have realised that in documenting the package I have not written or linked any style guide, as I was not sure how much contribution I would get. But since you are including code, it is important that we agree on certain style, so all the code within the package looks the same and we don't confuse users (and ourselves).

I am aware that for R there is no global agreement on coding standards, but [tidyverse coding standard](https://style.tidyverse.org/) seems to be one of the most widely accepted ones, and it's the one I am following.

So I will make some comments on your pull request according to this, mostly. I will also include this info in the README.

File: elsasserlib/R/normGeneCounts.R [hunk 1]
```diff
@@ -0,0 +1,48 @@
+
+#' Quantile Normalization of Gene Count Matrix
+#'
+#' This function replaces original values with estimated values according 
+#' to the rank within each sample.
+#'
+#' @param gene.counts Gene count table for normalization
+#' @return Normalised Counts
+#' @export
+quantileNorm <- function(gene.counts)
```
Comment (cnluzon) [inline elsasserlib/R/normGeneCounts.R]: `snake_case` styling is preferred for naming: `quantile_norm`. This is of course a style decision, but consistency is relevant.

File: elsasserlib/R/normGeneCounts.R [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+{
```
Comment (cnluzon) [inline elsasserlib/R/normGeneCounts.R]: This opening `{` should be in the same line as the function signature.

File: elsasserlib/R/normGeneCounts.R [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+#' Size Factor Normalization of Gene Count Matrix
+#'
+#' This function devides original values with size factors of each sample according 
+#' to DESeq method.
+#'
+#' @param gene.counts Gene count table for normalization
+#' @return Normalised Counts
+#' @export
+sizeFactorNorm <- function(gene.counts)
```
Comment (cnluzon) [inline elsasserlib/R/normGeneCounts.R]: same here: `size_factor_norm` is preferred, and `{` in the same line.

File: elsasserlib/R/normGeneCounts.R [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+#' @param gene.counts Gene count table for normalization
+#' @return Normalised Counts
+#' @export
+mrnNorm <- function(gene.counts)
+{ #  normalization
+  prenorm_mat <- t(t(gene.counts) / colSums(gene.counts))
+  tpm_sf <- colMedians(prenorm_mat / prenorm_mat[, 1])
+  norm_mat <- sweep(gene.counts, 2, tpm_sf / exp(mean(log(tpm_sf))), "/")
+  return(norm_mat)
+}
```
Comment (cnluzon) [inline elsasserlib/R/normGeneCounts.R]: missing end line at the end of file.

File: elsasserlib/R/normGeneCounts.R [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+#' Median Ratio Normalization of Gene Count Matrix
+#'
+#' This function devides original values with median ratios comparing to the first sample 
+#' according to edgeR method.
+#'
+#' @param gene.counts Gene count table for normalization
+#' @return Normalised Counts
+#' @export
+mrnNorm <- function(gene.counts)
```
Comment (cnluzon) [inline elsasserlib/R/normGeneCounts.R]: here, I would have a more self-explanatory name for the function: `mean_ratio_norm`, unless it is some kind of naming standard?

Also, `{` at the same line.

**Review by cnluzon — COMMENTED** [review]
In general just some style comments to agree on. Also, pull request titles should have some relation to the code included. Like: functions for gene count normalization or something like this. This way, changes are easier to track.

Thanks for contributing!

**Comment by shaorray** [comment]
Ok, I replaced the function name, and camel naming style can leave for methods.

I don't know how to compile the package to include the new file, roxygen2 doesn't add the new namespace to the package.

**Comment by cnluzon** [comment]
In general you should have the same style everywhere in your code, and even if it feels as nitpicking, it helps in the long run, and also cooperating with others. These are just common good programming practices.

Almost every respectable coding project that involves more than one person (and sometimes even those) has a style guide that developers follow. Some of the initial decisions are a matter of taste, but the code gains in quality and readability just by being consistent, it is definitely a bad practice to have mixed naming conventions within the same code.

You can also use `styler` to format code, it seems to be working nicely. 

About the compiling: If you are using R studio and open the R project attached to the package: It is my understanding that R studio will update the `NAMESPACE` file when you document or `devtools::build()`, I think. You should see some line that says `Writing NAMESPACE` in the process, and then if you look at the file you should see in there lines that contain `export(your_function)`. But *do not edit this* by hand, it should be generated by Roxygen2.

At this moment if vignettes are built it takes a bit long because there is some `bw` file that is processed. I'm working in including a small toy dataset that does not stall the build.

**Comment by cnluzon** [comment]
I have cloned your branch and it did work for me. So what I did:

- Open the project: `elsasserlib.Rproj` with R studio.
- Run documentation: <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>D</kbd> (I guess in Mac will be  <kbd>Command</kbd> + <kbd>Shift</kbd> + <kbd>D</kbd>). This writes some markdown `Md` documents needed so documentation shows when you hit `?quantile_norm`. 
- Load: <kbd>Command</kbd> + <kbd>Shift</kbd> + <kbd>L</kbd>.

Then you should be able to run it.

To build, you can call `devtools::build()` or <kbd>Command</kbd> + <kbd>Shift</kbd> + <kbd>B</kbd>. `devtools::check()` is also good to see if everything is ok: http://r-pkgs.had.co.nz/check.html

**Pull request merged**